tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+        Values greater than 1 will result in superior predictive performance, especially on smaller problems and with stacking enabled (reduces overall variance).
+        """
+        assert num_bag_sets > 0, 'num_bag_sets must be greater than zero'
+        self.config['num_bag_sets'] = num_bag_sets
+        return self
+
+    def num_stack_levels(self, num_stack_levels: int) -> ConfigBuilder:
+        """
+        Number of stacking levels to use in stack ensemble. Roughly increases model training time by factor of `num_stack_levels+1` (set = 0 to disable stack ensembling).
+        Disabled by default (0), but we recommend values between 1-3 to maximize predictive performance.
+        To prevent overfitting, `num_bag_folds >= 2` must also be set or else a ValueError will be raised.
+        """
+        assert num_stack_levels >= 0, 'num_stack_levels must be greater or equal than zero'
+        self.config['num_stack_levels'] = num_stack_levels
+        return self
+
+    def holdout_frac(self, holdout_frac: float) -> ConfigBuilder:
+        """
+        Fraction of train_data to holdout as tuning data for optimizing hyperparameters (ignored unless `tuning_data = None`, ignored if `num_bag_folds != 0` unless `use_bag_holdout == True`).
+        Default value (if None) is selected based on the number of rows in the training data. Default values range from 0.2 at 2,500 rows to 0.01 at 250,000 rows.
+        Default value is doubled if `hyperparameter_tune_kwargs` is set, up to a maximum of 0.2.
+        Disabled if `num_bag_folds >= 2` unless `use_bag_holdout == True`.
+        """
+        assert (holdout_frac >= 0) & (holdout_frac <= 1), 'holdout_frac must be between 0 and 1'
+        self.config['holdout_frac'] = holdout_frac
+        return self
+
+    def use_bag_holdout(self, use_bag_holdout: bool = True) -> ConfigBuilder:
+        """
+        If True, a `holdout_frac` portion of the data is held-out from model bagging.
+        This held-out data is only used to score models and determine weighted ensemble weights.
+        Enable this if there is a large gap between score_val and score_test in stack models.
+        Note: If `tuning_data` was specified, `tuning_data` is used as the holdout data.
+        Disabled if not bagging.
+        """
+        self.config['use_bag_holdout'] = use_bag_holdout
+        return self
+
+    def hyperparameter_tune_kwargs(self, hyperparameter_tune_kwargs: Union[str, dict]) -> ConfigBuilder:
+        """
+        Hyperparameter tuning strategy and kwargs (for example, how many HPO trials to run).
+        If None, then hyperparameter tuning will not be performed.
+        Valid preset values:
+            'auto': Uses the 'random' preset.
+            'random': Performs HPO via random search using local scheduler.
+        The 'searcher' key is required when providing a dict.
+        """
+        valid_str_values = ['auto', 'random']
```

Comment:
'bayesopt' should be a valid value, it is converted to 'random' downstream with a warning to avoid breaking user code based on prior releases.

Reply:
it is not: it's explicitly checked if the key is in the map keys: `autogluon.core.scheduler.scheduler_factory.get_hyperparameter_tune_kwargs_preset()`. And here are the valid keys:
```python
_scheduler_presets = {
    'auto': {'scheduler': 'local', 'searcher': 'local_random'},
    'local_random': {'scheduler': 'local', 'searcher': 'local_random'},
    'random': {'scheduler': 'local', 'searcher': 'random'},
}
```

Reply:
Good catch, can you fix in this PR by adding a new preset:

```
'bayesopt': {'scheduler': 'local', 'searcher': 'bayesopt'},
```